packages/image/src/Image.svelte

```diff
@@ -0,0 +1,198 @@
+				// > takes significantly more data. Even true 3x resolution screens are
+				// > wasteful as the human eye cannot see that level of detail without
+				// > something like a magnifying glass.
+				// https://blog.twitter.com/engineering/en_us/topics/infrastructure/2019/capping-image-fidelity-on-ultra-high-resolution-devices.html
+				[width, width * 2].map(
+					(w) => all_sizes.find((p) => p >= w) || all_sizes[all_sizes.length - 1]
+				)
+			)
+		];
+		return { widths, kind: 'x' };
+	}
+
+	/**
+	 * @param {string} src
+	 */
+	function matches_domain(src) {
+		const url = new URL(src, 'http://n'); // if src is protocol relative, use dummy domain
+		if (url.href === src) {
+			return domains.some((domain) => url.hostname === domain);
+		} else {
+			return true; // relative urls are always ok
+		}
+	}
+</script>
+
+<script>
+	/** @type {string | import('vite-imagetools').Img | import('vite-imagetools').Picture} */
+	export let src;
+	/** @type {string} */
+	export let alt;
+	/** @type {number}*/
+	export let width;
+	/** @type {number}*/
+	export let height;
+	/** @type {string | undefined} */
+	export let sizes = undefined;
+	/** @type {string | undefined} */
+	export let style = undefined;
+	/** @type {string} */
+	export let provider = 'default';
+	/** @type {any} */
+	export let providerOptions = undefined;
+	/** @type {'lazy' | 'eager' | undefined} */
+	export let loading = undefined;
+	/** @type {boolean} */
+	export let priority = false;
+
+	/** @type {import('vite-imagetools').Picture['sources'] | undefined} */
+	let picture_sources;
+	/** @type {string | undefined} */
+	let srcset;
+	let _src = '';
+
+	$: {
+		if (typeof src !== 'string') {
+			picture_sources = /** @type {import('vite-imagetools').Picture} */ (src).sources;
+			const img_src =
+				/** @type {import('vite-imagetools').Picture} */ (src).img ??
+				/** @type {import('vite-imagetools').Img} */ (src);
+			srcset = /** @type {import('vite-imagetools').Img} */ (src).srcset
+				?.map((i) => `${i.src} ${i.w}w`)
+				?.join(', ');
+			_src = img_src.src;
+			width = width || img_src.w;
+			height = height || img_src.h;
+		} else {
+			if (DEV && (!width || !height)) {
+				console.error(
+					`@sveltejs/image: You must provide a width and height for image with src '${src}'. Not providing these will cause layout shift which harms UX and SEO.`
+				);
+			}
+
+			if (matches_domain(src)) {
+				const p = providers[provider];
+				if (DEV && !p) {
+					throw new Error(
+						`@sveltejs/image: No provider named "${provider}" found. Available providers are ${Object.keys(
+							providers
+						).join(', ')}.`
+					);
+				}
+
+				const widths = getWidths(width, !!style && /width: \d+(px|em|rem)/.test(style), sizes);
+				srcset = widths.widths
+					.map((w) => {
+						const url = DEV
+							? src
+							: p.getURL({
+									src: /** @type {string} */ (src),
+									width: w,
+									options: providerOptions
+							  });
```

Comment:
why is `DEV` considered here? I kinda feel like it should be up to the provider to determine dev-time behaviour — e.g. if I'm using `vc dev` instead of `pnpm dev`, I _think_ `/_vercel/image` should work, and it would be preferable to see the optimised image at dev time

```suggestion
						const url = p.getURL({
							src: /** @type {string} */ (src),
							width: w,
							options: providerOptions
						 });
```